src/rdiff_backup/log.py

```diff
@@ -154,8 +146,8 @@ def log_to_file(self, message):
         if self.log_file_open:
             if self.log_file_local:
                 tmpstr = self.format(message, self.verbosity)
-                if type(tmpstr) != str:  # transform bytes into string
-                    tmpstr = str(tmpstr, 'utf-8')
+                if type(tmpstr) == str:  # transform string in bytes
+                    tmpstr = tmpstr.encode('utf-8', 'backslashreplace')
```

Comment:
I didn't know about `sys.stdout.encoding` but why not use it here as well? I truly hesitate because `cat logfile` would look better for the user if logfile and stdout have the same encoding, on the other hand, if a user sends us a logfile, it might be easier to check if it's always in Unicode. I would decide in the sense of the user and put same encoding on stdout and in the logfile, but I'll leave it to you, you're one of our bigger users :smiley:

Reply:
I have a hard time choosing the encoding for the file. I think it's more predictable if we force the encoding into 'utf-8'. 99% of the Linux platform are UTF-8. Mac OS X is using UTF8 and lately you have enforce the use of UTF-8 for windows too.

So I will stick with UTF8, on the long run it should not change anything for most of our users and will make the logging predictable for debugging.

Reply:
OK. Resolved.